### ISSUE TITLE
bundle packages refuses to save bundler itself

### ISSUE THREAD

**Issue by jesperronn** [rubygems/rubygems#3262]
Hey, I have an offline installation of ruby and required ruby gems where I ran into a problem.

I am preparing the installation on an online server, and then copy all the gems to the offline production server.

I used `bundle package --all` to create a `vendor/cache` folder containing all the gems I need. 

Unfortunately Bundler itself is missing. I am adding Bundler gem to the Gemfile to force it, but it did not work for me. 

Is this a bug, or is there a way, forcing it to save bundler as well in the cache?

Example project:

Gemfile:
```
source "https://rubygems.org"

gem "bundler"
```

Command:

```
$ bundle update --bundler
Using bundler 2.1.2
Warning: the lockfile is being updated to Bundler 2, after which you will be unable to return to Bundler 1.
Bundle updated!


$ bundle package
Using bundler 2.1.2
Bundle complete! 1 Gemfile dependency, 1 gem now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
Updating files in vendor/cache

$ ls -la vendor/cache
total 0
```

I prefer if Bundler can help me maintain all ruby dependencies. I realize I have to install it myself on the offline server to make `bundle install` work, but I still prefer the simple maintenance of putting bundler into the gemfile.

What would be the best current option for me?

**Comment by jeromedalbert** [comment]
I believe that this shouldn't be an issue any more, because Bundler has been included in Ruby since version 2.6 which came out in December 2018. What do you think @jesperronn @deivid-rodriguez?

**Comment by deivid-rodriguez** [comment]
I guess it'd be nice if the locked version of Bundler was cached, but I'm not sure how easy it's going to be. Let's see if @jesperronn still finds this inconvenient now that Bundler is provided by default.

**Comment by deivid-rodriguez** [comment]
Yeah, I imagined that was the case. I think it's a reasonable scenario, so I'll keep this open in case I can take a stab at it.

**Comment by jeromedalbert** [comment]
Just going on a limb here, but maybe using `bundle install --local` would solve your issue once https://github.com/rubygems/rubygems/issues/5227 is fixed via https://github.com/rubygems/rubygems/pull/8041 that David just opened. To my understanding it is supposed to only install from the Rubygems cache or `vendor/cache` and not connect to the internet.

**Comment by GigabyteProductions** [comment]
Regarding the combination of this issue and #5227:

I'm finding (with bundler 2.5.9, included in ruby 3.3.2) that if the locked bundler version is manually copied into `vendor/cache`, bundler is able to install the newer/locked version on an offline system when calling `bundler install --local` with the ruby-included version. Including bundler in the cache prevents the ruby-included one from trying to reach out to the internet (which addresses #5227), and avoids an issue where `bundler install --local` falls back to the non-locked version due to lack of internet connectivity.

With this, maybe #8041 should be reverted in favor of `bundler package` caching the bundler gem itself?

**Comment by deivid-rodriguez** [comment]
I want to implement caching the `bundler` gem itself as this issue suggests. Even if things work without it, it's best to use the locked version, so packaging it by default makes sense to me. And indeed, once that's implemented, #8041 will probably need tweaks. I don't think we want it completely reverted because if you don't have bundler in cache, we still want to respect `--local` and don't touch the network.
